File: Digital_Joper_mala/app/src/main/java/com/mala/digital_joper_mala/Helper/CacheHelper.java
package com.mala.digital_joper_mala.Helper;

import android.content.Context;
import android.content.SharedPreferences;

public class CacheHelper {

    private Context context;
    private String preference;
    private SharedPreferences sp;

    public CacheHelper(Context context, String preference) {
        this.context = context;
        this.preference = preference;
    }

    public void setCache(String key, String value){

        sp = context.getSharedPreferences(preference, Context.MODE_PRIVATE);
        sp.edit().putString(key, value).apply();

    }

    public String getCache(String key){

        sp = context.getSharedPreferences(preference, Context.MODE_PRIVATE);
        String values = sp.getString(key, "");

        return values;
    }

    public void deleteCache(String key){

        sp = context.getSharedPreferences(preference, Context.MODE_PRIVATE);
        sp.edit().remove(key).apply();

    }

}

File: Digital_Joper_mala/app/src/main/java/com/mala/digital_joper_mala/Helper/CacheHelper_.kt
package com.mala.digital_joper_mala.Helper

import android.content.Context
import android.content.SharedPreferences
import androidx.core.content.edit

class CacheHelper_(
    private val context: Context,
    private val preference : String
) {

    private var sp : SharedPreferences? = null

    fun setCache(key : String, value : String){

        sp = context.getSharedPreferences(preference, Context.MODE_PRIVATE)
        sp?.edit { putString(key, value) }

    }

    fun getCache(key: String, defaultValue : String) : String{

        sp = context.getSharedPreferences(preference, Context.MODE_PRIVATE)
        val values = sp?.getString(key, defaultValue)

        return values!!

    }

    fun deleteCache(key: String){

        sp = context.getSharedPreferences(preference, Context.MODE_PRIVATE)
        sp?.edit { remove(key) }

    }

}//class==============